plat/kvm/shutdown.c

```diff
@@ -26,12 +26,45 @@
 #include <uk/plat/common/irq.h>
 #include <uk/print.h>
 #include <uk/plat/bootstrap.h>
+#include <kvm/efi.h>
+#include <uk/plat/common/bootinfo.h>
 
 static void cpu_halt(void) __noreturn;
 
-/* TODO: implement CPU reset */
-void ukplat_terminate(enum ukplat_gstate request __unused)
+#ifdef CONFIG_KVM_BOOT_EFI_STUB
+static void uk_efi_rs_reset_system(uk_efi_reset_type_t reset_type)
```

Comment:
What if UEFI does not provide Runtime Services?

Reply:
Is that even possible? It would be as if you would not be able to do system resets, bios updates or manipulate firmware database variables. Nevertheless, I should definetely check for a `NULL` pointer as good practice anyway indeed.